### PR TITLE
1.18: add patch for reducing race risk in kubelet for missing KUBERNE…

### LINKED ISSUE #105: reduce race risk in kubelet for missing KUBERNETES_SERVICE_HOST 1.19+ fixed
### What happened?

https://github.com/kubernetes/kubernetes/issues/40973
https://github.com/kubernetes/kubernetes/issues/96070
https://github.com/kubernetes/kubernetes/pull/91500/files

### What did you expect to happen?

重启 kubelet 不影响pod

### How can we reproduce it (as minimally and precisely as possible)?

1.18 
systemclt restart kubelet

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
cherry pick from #91500 which is a fix in 1.19.

#### Which issue(s) this PR fixes:

Fixes #105

#### Does this PR introduce a user-facing change?

```release-note
1.18: add patch for reducing race risk in kubelet for missing KUBERNETES_SERVICE_HOST
```

### CODE REVIEW

File: releases.yml [hunk 1]
```diff
@@ -525,3 +526,7 @@ patches:
   - name: fix-etcd-put-key.1.11
     patch:
       - patches/fix-etcd-put-key.1.11.patch
+
+  - name: fix-missing-env-91500.1.18
```
Comment (wzshiming) [inline releases.yml:530]:
```suggestion

  # Pick commit from https://github.com/kubernetes/kubernetes/pull/91500.patch
  - name: fix-missing-env-91500.1.18
```

**Review by wzshiming — CHANGES_REQUESTED** [review]
https://github.com/klts-io/kubernetes-lts/runs/4663575052?check_suite_focus=true
``` console
+++ Conflicts detected:

UU pkg/kubelet/kubelet_pods_test.go
UU pkg/kubelet/kubelet_test.go
```

Follow the document to resolve the conflict https://github.com/klts-io/website/blob/main/content/zh/docs/addpatch/add-patch.md

**Comment by pacoxu** [comment]
```
+++ Returning you to the tag-v1.18.20 branch and cleaning up.
diff -a patches/fix-missing-env-91500.1.18.patch patches.bak/fix-missing-env-91500.1.18.patch
1c1
< From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
---
> From a7a9fbcd082c42175870d6ebe6d5b89d87ef8747 Mon Sep 17 00:00:00 2001
```
Fix the conflict.

**Comment by pacoxu** [comment]
```
W1230 10:30:42.693976   37451 mutation_detector.go:53] Mutation detector is enabled, this will result in memory leakage.
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x2b611a4]

goroutine 253 [running]:
testing.tRunner.func1(0xc000740400)
	/usr/local/go/src/testing/testing.go:874 +0x3a3
panic(0x3112400, 0x59cdb10)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/kubernetes/pkg/kubelet.(*Kubelet).makeEnvironmentVariables(0xc00144a000, 0xc00036d400, 0xc000c89340, 0x3777fb5, 0x7, 0xc0003dbf10, 0x1, 0x1, 0x4165370, 0xc000baeea0, ...)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_pods.go:578 +0x25d4
k8s.io/kubernetes/pkg/kubelet.TestMakeEnvironmentVariables.func1(0xc000740400)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/pkg/kubelet/kubelet_pods_test.go:1692 +0x616
testing.tRunner(0xc000740400, 0xc000d2e990)
	/usr/local/go/src/testing/testing.go:909 +0xc9
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:960 +0x350
FAIL	k8s.io/kubernetes/pkg/kubelet	5.439s
```
npe 再看看

**Comment by pacoxu** [comment]
image

**Pull request merged**